adapters/handlers/grpc/server.go

```diff
@@ -312,7 +324,30 @@ func extractPropertiesAnswer(results map[string]interface{}, properties search.S
 	return &props, nil
 }
 
-func extractPropertiesRequest(reqProps *pb.Properties) []search.SelectProperty {
+func getAllNonRefNonBlobProperties(scheme schema.Schema, className string) ([]search.SelectProperty, error) {
+	var props []search.SelectProperty
+	class := scheme.GetClass(schema.ClassName(className))
+
+	for _, prop := range class.Properties {
+		dt, err := schema.GetPropertyDataType(class, prop.Name)
+		if err != nil {
+			return []search.SelectProperty{}, err
+		}
+		if *dt == schema.DataTypeCRef || *dt == schema.DataTypeBlob {
+			continue
+		}
+
+		props = append(props, search.SelectProperty{
+			Name:        prop.Name,
+			IsPrimitive: true,
+		})
+
+	}
+
+	return props, nil
+}
+
+func extractPropertiesRequest(reqProps *pb.Properties, scheme schema.Schema, className string) []search.SelectProperty {
```

Comment:
Also, was `scheme` the intended spelling for the schema parameter?

Reply:
> Also, was scheme the intended spelling for the schema parameter?

Yes, so it doesn't collide with the schema package name :)

> Should this be extractRequestProperties?

it mirrors `extractPropertiesAnswer` that is used for the "trip back". But it might be time to split the file up into input and output